Clean/GadgetsNew/Addition32Full.lean

```diff
@@ -0,0 +1,119 @@
+import Mathlib.Algebra.Field.Basic
+import Mathlib.Data.ZMod.Basic
+import Clean.Utils.Primes
+import Clean.Utils.Vector
+import Clean.Circuit.Expression
+import Clean.Circuit.Provable
+import Clean.Circuit.Basic
+import Clean.Utils.Field
+import Clean.GadgetsNew.ByteLookup
+import Clean.GadgetsNew.Boolean
+import Clean.GadgetsNew.Add8.Addition8FullCarry
+import Clean.Types.U32
+
+namespace Addition32Full
+variable {p : ℕ} [Fact (p ≠ 0)] [Fact p.Prime]
+variable [p_large_enough: Fact (p > 512)]
+
+open Circuit
+open Provable (field field2 fields)
+open ByteLookup
+open Expression
+
+structure InputStruct (F : Type) where
+  x: U32 F
+  y: U32 F
+  carry_in: F
+
+def Inputs (p : ℕ) : TypePair := ⟨
+  InputStruct (Expression (F p)),
+  InputStruct (F p)
+⟩
+
+instance : ProvableType (F p) (Inputs p) where
+  size := 9 -- 4 + 4 + 1
+  to_vars s := vec [s.x.x0, s.x.x1, s.x.x2, s.x.x3, s.y.x0, s.y.x1, s.y.x2, s.y.x3, s.carry_in]
```

Comment:
note for later: if we implemented `Append` for `Vector` we might be able to compose these, like
```lean
ProvableType.to_vars s.x ++ ProvableType.to_vars s.y ++ ProvableType.to_vars s.carry_in
```